hazardous/data/_seer.py

```diff
@@ -0,0 +1,213 @@
+    "CS tumor size (2004-2015)",
+    "Total number of benign/borderline tumors for patient",
+    "Total number of in situ/malignant tumors for patient",
+]
+
+# "xxx" names are placeholders, these columns are not used.
+INDICES_TO_COLUMN_NAMES = {
+    0: "Patient ID",
+    1: "Sex",
+    2: "Year of diagnosis",
+    3: "Site recode ICD-0-3/WHO 2008",
+    4: "COD to site recode",
+    5: "Race recode (W, B, AI, API)",
+    6: "Histologic Type ICD-O-3",
+    7: "ICD-O-3 Hist/behav, malignant",
+    8: "Laterality",
+    9: "Sequence number",
+    10: "Vital status recode (study cutoff used)",
+    11: "ER Status Recode Breast Cancer (1990+)",
+    12: "PR Status Recode Breast Cancer (1990+)",
+    13: "Regional nodes examined (1988+)",
+    14: "xxx",
+    15: "Summary stage 2000 (1998-2017)",
+    16: "xxx",
+    17: "Reason no cancer-directed surgery",
+    18: "CS tumor size (2004-2015)",
+    19: "First malignant primary indicator",
+    20: "Diagnostic Confirmation",
+    21: "Total number of benign/borderline tumors for patient",
+    22: "Total number of in situ/malignant tumors for patient",
+    23: "Median household income inflation adj to 2019",
+    24: "RX Summ--Surg Prim Site (1998+)",
+    25: "SEER other cause of death classification",
+    26: "SEER cause-specific death classification",
+    27: "Survival months",
+    28: "xxx",
+}
+
+
+COLUMN_NAMES_TO_DTYPE = {
+    "Histologic Type ICD-O-3": "int64",
+}
+
+
+def load_seer(input_path, rename=True):
+    """Load and preprocess the seer dataset (expected as txt file).
+
+    This function reproduces the preprocessing heuristics from SurvTRACE:
+    https://github.com/RyanWangZf/SurvTRACE/blob/main/data/process_seer.py.
+
+    TODO: Write instructions to create the SEER file or link the SurvTRACE README.
+
+    Parameters
+    ----------
+    input_path : str, file_path or any valid input for ``pd.read_csv``.
+    """
+    if rename:
+        df = pd.read_csv(input_path, sep="\t", header=None)
+        df = add_column_names(df)
+    else:
+        df = pd.read_csv(input_path)
+
+    mask = (
+        (df["SEER cause-specific death classification"] != "N/A not seq 0-59")
+        & (
+            df["Reason no cancer-directed surgery"]
+            != "Not performed, patient died prior to recommended surgery"
+        )
+        & (df["Survival months"] != "Unknown")
+    )
+    df = df.loc[mask]
+
+    df_features = preprocess_features(df)
+    df_events = preprocess_events(df)
+
+    df_features = df_features.astype(COLUMN_NAMES_TO_DTYPE)
+
+    return df_features, df_events
+
+
+def add_column_names(X):
+    """Name inplace the columns of the SEER dataset using a column mapping.
+
+    Parameters
+    ----------
+    X : pandas.DataFrame of shape (n_features, n_samples)
+
+    Returns
+    -------
+    X : pandas.DataFrame of shape (n_features, n_samples)
+    """
+    X.rename(INDICES_TO_COLUMN_NAMES, axis=1, inplace=True)
+    return X
+
+
+def preprocess_features(X):
```

Comment:
Personally I think this kind of things should not be part of a the dataset loader but instead in the user's code (e.g. in examples or tests), by leveraging the rare categories options of scikit-learn or skrub categorical value encoders. But if we want to keep an option in the dataset loader to ease reproducibility of the paper, let's use a more explicit name such as:

```suggestion
def preprocess_seer_features_as_survtrace(X):
```

Reply:
Makes sense, thanks!